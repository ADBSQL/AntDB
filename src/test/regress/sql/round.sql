set grammar to oracle;
select round(2.009,2) from dual;
select round(100.987,0) from dual;
select round(100.987,-1) from dual;
select round(100.987,-3) from dual;
select round(exp(2),2) from dual;
select round(100.984,'2') from dual;
select round('100.984','2') from dual;
select round(100.984,2.8) from dual;
select round(100.984,-2.8) from dual;
select round(-100.98467,3) from dual;
select round(1000000000000000000,3) from dual;
select round(100,50) from dual;
select round(sinh(5),3) from dual;
select round(sinh(5),null) from dual;
select round(cosh(3),'') from dual;