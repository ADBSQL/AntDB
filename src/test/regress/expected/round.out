set grammar to oracle;
select round(2.009,2) from dual;
 round 
-------
  2.01
(1 row)

select round(100.987,0) from dual;
 round 
-------
   101
(1 row)

select round(100.987,-1) from dual;
 round 
-------
   100
(1 row)

select round(100.987,-3) from dual;
 round 
-------
     0
(1 row)

select round(exp(2),2) from dual;
 round 
-------
  7.39
(1 row)

select round(100.984,'2') from dual;
 round  
--------
 100.98
(1 row)

select round('100.984','2') from dual;
ERROR:  not a valid month: "984"
select round(100.984,2.8) from dual;
 round  
--------
 100.98
(1 row)

select round(100.984,-2.8) from dual;
 round 
-------
   100
(1 row)

select round(-100.98467,3) from dual;
  round   
----------
 -100.985
(1 row)

select round(1000000000000000000,3) from dual;
          round          
-------------------------
 1000000000000000000.000
(1 row)

select round(100,50) from dual;
                         round                          
--------------------------------------------------------
 100.00000000000000000000000000000000000000000000000000
(1 row)

select round(sinh(5),3) from dual;
 round  
--------
 74.203
(1 row)

select round(sinh(5),null) from dual;
 round 
-------
      
(1 row)

select round(cosh(3),'') from dual;
 round 
-------
      
(1 row)

